src/Xamarin.Android.Tools.AndroidSdk/Sdks/AndroidSdkUnix.cs

```diff
@@ -158,6 +166,10 @@ public override void SetPreferredAndroidNdkPath (string? path)
 			path = NullIfEmpty (path);
 
 			var doc = GetUnixConfigFile (Logger);
+
+			if (doc.Root is null)
+				throw new InvalidOperationException ("Unix config file root is missing");
```

Comment:
Should this include the path that couldn't be found?  If this exception "leaks" to the user, it's not particularly "actionable".

Reply:
What's "doubly hilarious" is that this codepath should be unreachable, i.e. *it isn't possible for `doc.Root` to be `null`*:

https://github.com/xamarin/xamarin-android-tools/blob/7ef8ec51f4ca2ddd2f651b87b487c367ac833ceb/src/Xamarin.Android.Tools.AndroidSdk/Sdks/AndroidSdkUnix.cs#L254-L256

`GetUnixConfigFile()` *ensures* that `doc.Root` isn't `null`!

And you've had to "duplicate" this check in multiple places.

Perhaps the "better" solution is to change the return type of `GetUnixConfigFile()` to `XElement` instead of `XDocument`, and have it be the "root" element?

Reply:
and maybe update `SaveConfig()` to likewise take `XElement` instead of `XDocument`?

I'm not sure we *need* to use `XDocument` here…